resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -401,7 +401,13 @@ private[spark] class Client(
     if (force || !compareFs(srcFs, destFs) || "file".equals(srcFs.getScheme)) {
       destPath = new Path(destDir, destName.getOrElse(srcPath.getName()))
       logInfo(s"Uploading resource $srcPath -> $destPath")
-      FileUtil.copy(srcFs, srcPath, destFs, destPath, false, hadoopConf)
+      try {
+        FileUtil.copy(srcFs, srcPath, destFs, destPath, false, hadoopConf)
+      } catch {
+        // HADOOP-16878 changes the behavior to throw exceptions when src equals to dest
+        case e: PathOperationException
+            if force && srcFs.makeQualified(srcPath).equals(destFs.makeQualified(destPath)) =>
```

Comment:
should we just skip calling `copy` in this case?

Reply:
Do you mean `overwrite` in Spark side as a fallback?

Reply:
I mean don't call `FileUtil.copy` in the case, if we already know it will throw an exception.

Reply:
The exception depends on the underlying Hadoop versions. Technically, Apache Hadoop 3.3.1 also should not throw this. I suspect that something is messed up in Jenkins `.m2` directory.

Reply:
BTW, I don't want to introduce any behavior change here. As @HyukjinKwon 's pointed out, there might be a corner case in some cases (https://github.com/apache/spark/pull/32983#issuecomment-864492077).
> This doesn't seems to handle the case where the nameservice of both source & target filesystem is same, but they point to different clusters. So, it gives a false alarm in that case.